File: leetcode/41-first-missing-positive/missing_positive.py
"""
LC 41: First Missing Positive
Given an unsorted integer array nums. Return the smallest positive integer that is not present in nums.

You must implement an algorithm that runs in O(n) time and uses O(1) auxiliary space.

"""
class Solution:
    def first_missing_positive(self, nums: List[int]) -> int:
        nums.sort()
        number = 1

        for num in number:
            if num > 0 and num == number:
                number += 1
        return number


def main():
    sol = Solution.first_missing_positive([1, 2, 3, 4, 5])
    print(sol)


if __name == "__main__":
    main()

File: leetcode/41-first-missing-positive/MissingPositive.java
// LC 41: First Missing Positive

class MissingPositive {
    public int firstMissingPositive(int[] nums) {
        Arrays.sort(nums);
        int number = 1;

        for (int num : nums) {
            if (num > 0 && num == number)
                number++;
        }

        return number;
    }
}
